multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/RoutesValidator.java

```diff
@@ -74,22 +73,42 @@ public Object attemptToCorrect(Object routes, final Map<String, Object> context)
         }
 
         return routesList.stream()
-                         .map(route -> attemptToCorrectParameterMap(route, context))
+                         .map(routesElement -> getRelatedParametersForContext(routesElement, context))
+                         .map(this::attemptToCorrectParameterMap)
                          .collect(Collectors.toList());
     }
 
-    private Map<String, Object> attemptToCorrectParameterMap(Map<String, Object> originalElem, Map<String, Object> context) {
+    private Map<String, Object> getRelatedParametersForContext(Map<String, Object> routesElement, Map<String, Object> context) {
+        Map<String, Object> updatedContext = new HashMap<>(routesElement);
+        Set<String> relatedParameters = getRelatedParameterNames();
+        Map<String, Object> updatedParamters = relatedParameters.stream()
+                                                                .filter(relatedParameter -> doesParameterMissingInRoutesElement(relatedParameter,
+                                                                                                                                routesElement,
+                                                                                                                                context))
+                                                                .collect(Collectors.toMap(Function.identity(), context::get));
```

Comment:
is it guaranteed that these parameters are in the context? this way of processing does not specify a default value if they are not present.

Also this whole method is a bit hard to understand. Maybe also name 'updatedParameters' something more clear e.g. globalParameterValues

Reply:
About the first question the answer is yes, in `isGlobalParameterValueOverridden` method there is a check does context contains this parameter key. I have tried to reproduce scenario where the key is present but the value is null but this is not possible because when in yaml set value of key to null, then it is ignored.

Reply:
yes